bgpd/bgp_io.c

```diff
@@ -447,11 +447,17 @@ static uint16_t bgp_read(struct peer *peer)
 	size_t readsize; // how many bytes we want to read
 	ssize_t nbytes;  // how many bytes we actually read
 	uint16_t status = 0;
-	static uint8_t ibw[BGP_MAX_PACKET_SIZE * BGP_READ_PACKET_MAX];
+	uint8_t ibw[BGP_PACKET_SIZE(peer) * BGP_READ_PACKET_MAX];
 
-	readsize = MIN(ringbuf_space(peer->ibuf_work), sizeof(ibw));
+	readsize = MIN(ringbuf_space(peer->ibuf_work), BGP_PACKET_SIZE(peer));
 	nbytes = read(peer->fd, ibw, readsize);
 
+	if (BGP_DEBUG(update, UPDATE_IN))
```

Comment:
This should already happen in bgp_process_packets

Reply:
I'm sorry, didn't get your point here. Do you want me to remove this debug logging at all? Or how would you like to be here?

Reply:
First of all - `bgp_read` reads data off a stream socket. We don't even know if this is a BGP message yet - it could be straight garbage. Second, even if it is a BGP message we don't know if it's an UPDATE, so a debug guard used for logging UPDATE messages is the wrong thing to use. Third, if you want to actually debug UPDATEs, we already have code to do that in `bgp_packet.c`:
```
  1569         if (BGP_DEBUG(update, UPDATE_IN))
  1570                 zlog_debug("%s rcvd UPDATE wlen %d attrlen %d alen %d",
  1571                            peer->host, withdraw_len, attribute_len, update_len);
```
Now if your intent is to debug read events then maybe this is fine, but the debug guard needs to be changed at least.

Reply:
I decided to remove this logging at all. Too much chatty. This can be verified using tcpdump/tshark if needed.